features/datefields.feature

```diff
@@ -0,0 +1,64 @@
+Feature: Form submission with date fields
+
+    Background:
+        Given there is a Search API that gives a successful response and returns 10 results
+        And the search controller is running
+        And I navigate to "/alladhocs"
+
+    Scenario: Submitting the form with valid date fields
+        Given I fill in input element "#after-date-day" with value "01"
+        And I fill in input element "#after-date-month" with value "01"
+        And I fill in input element "#after-date-year" with value "2022"
+        And I fill in input element "#before-date-day" with value "31"
+        And I fill in input element "#before-date-month" with value "12"
+        And I fill in input element "#before-date-year" with value "2022"
+        When I click the "#search-filter > button" element
+        And I wait 1 seconds
+        Then element "#error-summary-title" should not be visible
+
+    Scenario Outline: Submitting the form with invalid date fields
+        Given I fill in input element "#after-date-day" with value "<day>"
+        And I fill in input element "#after-date-month" with value "<month>"
+        And I fill in input element "#after-date-year" with value "<year>"
+        When I click the "#search-filter > button" element
+        And I wait 1 seconds
+        Then element "#error-summary-title" should be visible
+        And element "#fromDate-error" should be visible
```

Comment:
I don't *think* you need the wait here as 'should be visible' has a built in wait, might be wrong though.